hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java

```diff
@@ -1293,4 +1300,144 @@ public void testCompactionDagDisableForSnapshotMetadata() throws Exception {
+      int oldestIndex, int latestIndex, int snapshotRetentionPeriod,
+      String volumeNameStr, String bucketNameStr, OzoneBucket ozoneBucketClient)
+      throws Exception {
+    if (latestIndex - oldestIndex >= snapshotRetentionPeriod) {
+      store.deleteSnapshot(volumeNameStr, bucketNameStr,
+          snapshotPrefix + oldestIndex);
+      oldestIndex++;
+    }
+    createFileKeyWithData(ozoneBucketClient, KEY_PREFIX + latestIndex,
+        KEY_TEST_DATA);
+    createSnapshot(volumeNameStr, bucketNameStr, snapshotPrefix + latestIndex);
+    latestIndex++;
+    return new int[]{oldestIndex, latestIndex};
+  }
+
+  private void checkDayWeekMonthSnapshotData(OzoneBucket ozoneBucketClient,
+                                             int latestDayIndex,
+                                             int latestWeekIndex,
+                                             int latestMonthIndex,
+                                             String keyTestData)
+      throws Exception {
+    String curKey = "";
+    for (int i = 0; i < latestDayIndex; i++) {
+      curKey = KEY_PREFIX + i;
+      //validate keys metadata
+      OzoneKeyDetails ozoneKeyDetails = ozoneBucketClient.getKey(curKey);
+      Assert.assertEquals(curKey, ozoneKeyDetails.getName());
+      Assert.assertEquals(ozoneBucketClient.getName(),
+          ozoneKeyDetails.getBucketName());
+      Assert.assertEquals(ozoneBucketClient.getVolumeName(),
+          ozoneKeyDetails.getVolumeName());
+
+      //validate keys data
+      try (OzoneInputStream ozoneInputStream =
+               ozoneBucketClient.readKey(curKey)) {
+        byte[] fileContent = new byte[keyTestData.length()];
+        ozoneInputStream.read(fileContent);
+        Assert.assertEquals(keyTestData, new String(fileContent, UTF_8));
+      }
+
+      //validate day snapshot data integrity
+      validateSnapshotDataIntegrity(SNAPSHOT_DAY_PREFIX, latestDayIndex, curKey,
+          ozoneBucketClient);
+      //validate week snapshot data integrity
+      validateSnapshotDataIntegrity(SNAPSHOT_WEEK_PREFIX, latestWeekIndex,
+          curKey, ozoneBucketClient);
+      //validate month snapshot data integrity
+      validateSnapshotDataIntegrity(SNAPSHOT_MONTH_PREFIX, latestMonthIndex,
+          curKey, ozoneBucketClient);
+    }
+
+  }
+
+  private void validateSnapshotDataIntegrity(String snapshotPrefix, int index,
+                                             String keyName,
+                                             OzoneBucket ozoneBucketClient)
+      throws Exception {
+    Iterator<? extends OzoneKey> iterator =
+        ozoneBucketClient.listKeys(
+            OmSnapshotManager.getSnapshotPrefix(snapshotPrefix + (index - 1)) +
+                KEY_PREFIX);
+    boolean findKey = false;
+    while (iterator.hasNext()) {
+      OzoneKey ozoneKey = iterator.next();
+      if (ozoneKey.getName().contains(keyName)) {
+        findKey = true;
+        break;
+      }
+    }
```

Comment:
In a follow-up patch, we should check the actual content written to the key as well.

Reply:
sure, updated